src/main/scala/monocle/Optic.scala

```diff
@@ -4,202 +4,184 @@ import monocle.impl._
 
-type Optic[+ThisCan, S, A]      = POptic[ThisCan, S, S, A, A]
-type OpticGet[+ThisCan, -S, +A] = POptic[ThisCan, S, Nothing, A, Any]
+type Optic[+Can, S, A]      = POptic[Can, S, S, A, A]
+type OpticGet[+Can, -S, +A] = POptic[Can, S, Nothing, A, Any]
 
-object GetOptic:
+object OpticGet:
   def id[A]: OpticGet[Get, A, A] =
     Optic.thatCan.get[A, A](identity)
 
 object Optic:
-  def id[A]: Optic[Get & ReverseGet, A, A] =
+  def id[A]: Optic[ConvertBetween, A, A] =
     POptic.id
 
   object thatCan extends OpticConstructors(POptic.thatCan)
 
 end Optic
 
 object POptic:
-  def id[A, B]: POptic[Get & ReverseGet, A, B, A, B] =
+  def id[A, B]: POptic[ConvertBetween, A, B, A, B] =
     thatCan.convertBetween[A, B, A, B](identity)(identity)
 
   object thatCan extends POpticConstructors
 
 end POptic
 
-final class POptic[+ThisCan, -S, +T, +A, -B] private[monocle] (
-  protected[monocle] val impl: OpticImpl[ThisCan, S, T, A, B]
+final class POptic[+Can, -S, +T, +A, -B] private[monocle] (
+  protected[monocle] val impl: OpticImpl[Can, S, T, A, B]
 ):
 
-  def andThen[ThatCan >: ThisCan, C, D](that: POptic[ThatCan, A, B, C, D]): POptic[ThatCan, S, T, C, D] =
-    POptic(impl.andThen(that.impl))
+  def andThen[Can2 >: Can, C, D](optic2: POptic[Can2, A, B, C, D]): POptic[Can2, S, T, C, D] =
+    POptic(impl.andThen(optic2.impl))
 
-  def index[I, S1 <: S, A1 >: A, V](i: I)(using
-    evIndex: Index[A1, I, V]
-  )(using T <:< S1, A1 <:< B): Optic[ThisCan | (GetOption & Modify), S1, V] =
-    asInstanceOf[Optic[ThisCan, S1, A1]].andThen(evIndex.index(i))
+  def all(p: A => Boolean)(using Can <:< GetMany): S => Boolean =
+    impl.safe.toIterator(_).forall(p)
 
-  override def toString() =
-    s"POptic(${impl})"
-
-end POptic
+  def choice[S1 >: T <: S, A1 >: A <: B, S2](other: Optic[Get, S2, A1])(using Can <:< Get): Optic[Get, Either[S1, S2], A1] =
+    Optic.thatCan.get(_.fold(this.get, other.get))
 
-extension [S, T, A, B](self: POptic[GetMany, S, T, A, B])
-  def all(p: A => Boolean): S => Boolean =
-    self.impl.safe.toIterator(_).forall(p)
+  def each[S1 >: T <: S, A1 >: A <: B, C](using evEach: Each[A1, C]): Optic[Can | EditMany, S1, C] =
+    andThen(evEach.each)
 
-extension [S, A](self: Optic[Get, S, A])
-  def choice[S1](other: Optic[Get, S1, A]): Optic[Get, Either[S, S1], A] =
-    Optic.thatCan.get(_.fold(self.get, other.get))
+  def exist(p: A => Boolean)(using Can <:< GetMany): S => Boolean =
+    impl.safe.toIterator(_).exists(p)
 
-extension [ThisCan, S, A](self: Optic[ThisCan, S, A])
-  def each[C](using evEach: Each[A, C]): Optic[ThisCan | (GetMany & Modify), S, C] =
-    self.andThen(evEach.each)
+  def find(p: A => Boolean)(using Can <:< GetMany): S => Option[A] =
+    impl.safe.toIterator(_).find(p)
 
-extension [S, T, A, B](self: POptic[GetMany, S, T, A, B])
-  def exist(p: A => Boolean): S => Boolean =
-    self.impl.safe.toIterator(_).exists(p)
+  def fold[A1 >: A : Monoid](s: S)(using Can <:< GetMany): A1 = 
```

Comment:
I think type aliases for `Can` combination complexifies the API because users need to learn about underlying `Can` flag in the methods.

Reply:
Actually, the intersection type is still visible in some methods like `andThen`